rust/src/domains/polars/series/mod.rs

```diff
@@ -0,0 +1,315 @@
+    NumericNative<PolarsType = Self::NumericPolars> + PrimitiveDataType
+{
+    type NumericPolars: PolarsDataType + PolarsNumericType<Native = Self>;
+}
+
+pub trait PrimitiveDataType: 'static + Send + Sync {
+    type Polars: PolarsDataType;
+    fn dtype() -> DataType {
+        Self::Polars::get_dtype()
+    }
+}
+
+macro_rules! impl_dtype_from {
+    ($ty:ty, $dt:ty) => {
+        impl NumericDataType for $ty {
+            type NumericPolars = $dt;
+        }
+        impl PrimitiveDataType for $ty {
+            type Polars = $dt;
+        }
+    };
+}
+impl_dtype_from!(u8, UInt8Type);
+impl_dtype_from!(u16, UInt16Type);
+impl_dtype_from!(u32, UInt32Type);
+impl_dtype_from!(u64, UInt64Type);
+impl_dtype_from!(i8, Int8Type);
+impl_dtype_from!(i16, Int16Type);
+impl_dtype_from!(i32, Int32Type);
+impl_dtype_from!(i64, Int64Type);
+impl_dtype_from!(f32, Float32Type);
+impl_dtype_from!(f64, Float64Type);
+impl PrimitiveDataType for bool {
+    type Polars = BooleanType;
+}
+impl PrimitiveDataType for String {
+    type Polars = StringType;
+}
+#[cfg(test)]
+mod test_series {
+    use crate::domains::OptionDomain;
+
+    use super::*;
+
+    #[test]
+    fn test_series_new() -> Fallible<()> {
+        let series_domain = SeriesDomain::new("A", AtomDomain::<bool>::default());
+        assert!(series_domain == series_domain);
+
+        let series = Series::new("A", vec![true; 50]);
+        assert!(series_domain.member(&series)?);
+        Ok(())
+    }
+
+    #[test]
+    fn test_series_bounded() -> Fallible<()> {
+        let series_domain = SeriesDomain::new("A", AtomDomain::new_closed((1, 3))?);
+
+        let inside_bounds = Series::new("A", vec![1; 50]);
+        assert!(series_domain.member(&inside_bounds)?);
+
+        let outside_bounds = Series::new("A", vec![4; 50]);
+        assert!(!series_domain.member(&outside_bounds)?);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_series_non_nullable() -> Fallible<()> {
+        // option domain with non-nullable type
+        let series_domain =
+            SeriesDomain::new("A", OptionDomain::new(AtomDomain::<bool>::default()));
+
+        let series = Series::new("A", vec![Some(true), Some(false), None]);
+        assert!(series_domain.member(&series)?);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_series_nullable_without_option() -> Fallible<()> {
+        // nullable type without options
+        let series_domain = SeriesDomain::new("A", AtomDomain::<f64>::new_nullable());
+
+        // None is not ok, but NaN is ok
+        let series = Series::new("A", vec![Some(1.), Some(f64::NAN), None]);
+        assert!(!series_domain.member(&series)?);
+        // series made with Option::Some are ok
+        let series = Series::new("A", vec![Some(1.), Some(f64::NAN)]);
+        assert!(series_domain.member(&series)?);
+        // series made without options are ok
+        let series = Series::new("A", vec![1., f64::NAN]);
+        assert!(series_domain.member(&series)?);
```

Comment:
```suggestion
        let series_with_none = Series::new("A", vec![Some(1.), Some(f64::NAN), None]);
        assert!(!series_domain.member(&series_with_none)?);
        
        // series made with Option::Some are ok
        let series_with_some = Series::new("A", vec![Some(1.), Some(f64::NAN)]);
        assert!(series_domain.member(&series_with_some)?);
        
        // series made without options are ok
        let series_wo_some = Series::new("A", vec![1., f64::NAN]);
        assert!(series_domain.member(&series_wo_some)?);
```

Where we really have separate tests that happen to be run serially, I think it improves readability to use distinct names, and add a line break between.